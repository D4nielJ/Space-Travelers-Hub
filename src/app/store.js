import {
  createStore, combineReducers, applyMiddleware, compose,
} from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';

const middlewares = [thunk, logger];
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const reducer = combineReducers({
  // missions: missionsReducer,
  // rockets: rocketsReducer
});

const store = createStore(
  reducer,
  composeEnhancers(applyMiddleware(...middlewares)),
);

export default store;