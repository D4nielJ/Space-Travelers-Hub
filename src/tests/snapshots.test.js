import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Rockets from '../features/rockets/rockets';
import Missions from '../features/missions/missions';
import Profile from '../features/profile/profile';
import store from '../app/store';

describe('Tests snapshots of components', () => {
  it('renders correctly', () => {
    const rocketsTree = renderer
      .create(
        <Provider store={store}>
          <Rockets />
        </Provider>,
      )
      .toJSON();
    expect(rocketsTree).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const missionsTree = renderer
      .create(
        <Provider store={store}>
          <Missions />
        </Provider>,
      )
      .toJSON();
    expect(missionsTree).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const profileTree = renderer
      .create(
        <Provider store={store}>
          <Profile />
        </Provider>,
      )
      .toJSON();
    expect(profileTree).toMatchSnapshot();
  });
});
