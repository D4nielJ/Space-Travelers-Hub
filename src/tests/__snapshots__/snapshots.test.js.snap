// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests snapshots of components renders correctly 1`] = `
<div
  className="rocketsContainer"
/>
`;

exports[`Tests snapshots of components renders correctly 2`] = `
<table
  className="border border-gray-400 text-left mb-16"
>
  <thead
    className="bg-gray-300"
  >
    <tr>
      <th
        className="border border-gray-400 py-2 px-2 text-lg text-bold"
      >
        Mission
      </th>
      <th
        className="border border-gray-400 py-2 px-2 text-lg text-bold"
      >
        Description
      </th>
      <th
        className="border border-gray-400 py-2 px-2 text-lg text-bold"
      >
        Status
      </th>
      <th
        className="border border-gray-400 py-2 px-2"
      >
        Â 
      </th>
    </tr>
  </thead>
  <tbody />
</table>
`;

exports[`Tests snapshots of components renders correctly 3`] = `
<section
  className="flex justify-around"
>
  <div
    className="w-1/2"
  >
    <h2
      className="font-semibold text-xl mb-4"
    >
      My Missions
    </h2>
    <ul
      className="border rounded"
    >
      <li
        className="p-6 font-medium"
      >
        Join missions to start
      </li>
    </ul>
  </div>
  <div
    className="w-1/2 ml-8"
  >
    <h2
      className="font-semibold text-xl mb-4"
    >
      My Rockets
    </h2>
    <ul
      className="border rounded"
    >
      <li
        className="p-6 font-medium"
      >
        Reserve a Rocket
      </li>
    </ul>
  </div>
</section>
`;
